pkg/crawler/crawler.go

```diff
@@ -5,31 +5,57 @@ import (
 	"encoding/hex"
 	"encoding/xml"
 	"fmt"
-	"github.com/aquasecurity/trivy-java-db/pkg/fileutil"
-	"github.com/aquasecurity/trivy-java-db/pkg/types"
+	"hash/fnv"
 	"io"
 	"log"
 	"net/http"
+	"os"
 	"path/filepath"
+	"sort"
 	"strings"
 	"sync"
+	"time"
+
+	"github.com/aquasecurity/trivy-java-db/pkg/fileutil"
+	"github.com/aquasecurity/trivy-java-db/pkg/types"
+	"github.com/google/licenseclassifier/v2/tools/identify_license/backend"
+	"github.com/samber/lo"
 
 	"github.com/PuerkitoBio/goquery"
 	"github.com/hashicorp/go-retryablehttp"
+	cmap "github.com/orcaman/concurrent-map/v2"
+	"golang.org/x/net/html/charset"
 	"golang.org/x/sync/semaphore"
 	"golang.org/x/xerrors"
 )
 
 const mavenRepoURL = "https://repo.maven.apache.org/maven2/"
 
 type Crawler struct {
-	dir  string
-	http *retryablehttp.Client
+	dir        string
+	licensedir string
+	http       *retryablehttp.Client
 
 	rootUrl string
 	wg      sync.WaitGroup
 	urlCh   chan string
 	limit   *semaphore.Weighted
+	opt     Option
+
+	// license classifier
+	classifier *backend.ClassifierBackend
+
+	// list of temporary license files
+	files []string
```

Comment:
Looks like we can return this slice in https://github.com/aquasecurity/trivy-java-db/blob/5ec049b1de6facc842d6a346fab72bb86c06df4b/pkg/crawler/crawler.go#L426

Reply:
Makes sense. Fixed